models/action.go

```diff
@@ -633,6 +634,61 @@ func CommitRepoAction(opts CommitRepoActionOptions) error {
 			}
 		}
 
+		// if there is a pull request fro this branch, then update pull request
+		prs, err := GetUnmergedPullRequestsByHeadInfo(repo.ID, refName)
+		if err != nil {
+			return fmt.Errorf("GetUnmergedPullRequestsByHeadInfo: %v", err)
+		}
+
+		if len(prs) > 0 {
+			var isForcePush = false
+
+			// detect force push
+			if !isNewBranch && git.EmptySHA != opts.Commits.Commits[0].Sha1 {
+				output, err := git.NewCommand("rev-list", opts.Commits.Commits[0].Sha1, "^"+opts.Commits.Commits[len(opts.Commits.Commits)-1].Sha1).RunInDir(repo.RepoPath())
+				if err != nil {
+					return fmt.Errorf("rev-list: %v", err)
+				} else if len(output) > 0 {
+					isForcePush = true
```

Comment:
I don't think this is right. Suppose I push 3 commits on top of a branch's current head (not a force push)
```
sha1 newest (my parent is sha2)
sha2 older (my parent is sha3)
sha3 even older (my parent is the previous head of the branch)
```

`opts.Commits.Commits` will contain `sha1`, `sha2` and `sha3`, in that order. We will run `git rev-list sha1 ^sha3`, which will return `sha1` and `sha2`, and this push will incorrectly be marked as a force push.

Reply:
I think it should be this:
`output, err := git.NewCommand("rev-list", opts.OldCommitID, "^"+opts.NewCommitID).RunInDir(repo.RepoPath())`

Reply:
done.